NBitcoin.Tests/NetworkAddressTests.cs

```diff
@@ -0,0 +1,567 @@
+
+			// Valid CJDNS.
+			payload = Encoders.Hex.DecodeData(
+				"00" +         // services (varint)
+				"06" +         // network type (CJDNS)
+				"10" +         // address length
+				"fc000001000200030004000500060007" + // address
+				"0000");       // port
+			stream = new BitcoinStream(payload){
+				ProtocolVersion = NetworkAddress.AddrV2Format,
+				Type = SerializationType.Hash
+			};
+			stream.ReadWrite(ref addr);
+			Assert.True(addr.Endpoint.IsValid());
+			Assert.True(addr.IsCjdns);
+			Assert.False(addr.IsAddrV1Compatible);
+			Assert.Equal("fc00:1:2:3:4:5:6:7", addr.ToAddressString());
+			Assert.True(stream.Inner.Length == stream.Inner.Position);
+
+			// Invalid CJDNS, with bogus length.
+			payload = Encoders.Hex.DecodeData(
+				"00" +         // services (varint)
+				"06" +         // network type (CJDNS)
+				"01" +         // address length
+				"00");         // address
+			stream = new BitcoinStream(payload){
+				ProtocolVersion = NetworkAddress.AddrV2Format,
+				Type = SerializationType.Hash
+			};
+			ex = Assert.Throws<ArgumentException>(() => stream.ReadWrite(ref addr));
+			Assert.Equal("BIP155 Cjdns address is 1 bytes long. Expected length is 16 bytes.", ex.Message);
+
+			// Unknown, with extreme length.
+			payload = Encoders.Hex.DecodeData(
+				"00" +         // services (varint)
+				"aa" +         // network type (unknown)
+				"fe00000002" + // address length (CompactSize's MAX_SIZE)
+				"01020304050607"); // address
+			stream = new BitcoinStream(payload){
+				ProtocolVersion = NetworkAddress.AddrV2Format,
+				Type = SerializationType.Hash
+			};
+			ex = Assert.Throws<ArgumentOutOfRangeException>(() => stream.ReadWrite(ref addr));
+			Assert.Contains("Array size too big", ex.Message);
+			Assert.True(stream.Inner.Length != stream.Inner.Position);
+
+			// Unknown, with reasonable length.
+			payload = Encoders.Hex.DecodeData(
+				"00" +         // services (varint)
+				"aa" +         // network type (unknown)
+				"04" +         // address length
+				"01020304" +   // address
+				"0000");       // port
+			stream = new BitcoinStream(payload){
+				ProtocolVersion = NetworkAddress.AddrV2Format,
+				Type = SerializationType.Hash
+			};
+			stream.ReadWrite(ref addr);
+			Assert.False(addr.Endpoint.IsValid());
+			Assert.True(stream.Inner.Length == stream.Inner.Position);
+
+			// Unknown, with zero length.
+			payload = Encoders.Hex.DecodeData(
+				"00" +         // services (varint)
+				"aa" +         // network type (unknown)
+				"00" +         // address length
+				"0000");       // port
+			stream = new BitcoinStream(payload){
+				ProtocolVersion = NetworkAddress.AddrV2Format,
+				Type = SerializationType.Hash
+			};
+			stream.ReadWrite(ref addr);
+			Assert.False(addr.Endpoint.IsValid());
+			Assert.True(stream.Inner.Length == stream.Inner.Position);
+		}
+
+		static NetworkAddress[] fixture_addresses = new NetworkAddress[]
```

Comment:
maybe `ipv6_fixture_addresses`?

Reply:
No, let it as it is. Renaming this to ipv6 fixture is more confusing because the very first test is for verifying the address is IPv4.